src/py/run_program.rs

```diff
@@ -28,6 +36,241 @@ pub fn operator_handler2(
     allocator.err(&op_arg, "unimplemented operator")
 }
 
+fn u32_from_u8(mut buf: &[u8]) -> Option<u32> {
+    // skip leading zeroes
+    while !buf.is_empty() && buf[0] == 0 {
+        buf = &buf[1..];
+    }
+
+    if buf.is_empty() {
+        return Some(0);
+    }
+
+    // too many bytes for u32
+    if buf.len() > 4 {
+        return None;
+    }
+
+    let mut ret: u32 = 0;
+    for b in buf {
+        ret <<= 8;
+        ret |= *b as u32;
+    }
+    Some(ret)
+}
+
+#[test]
+fn test_u32_from_u8() {
+    assert_eq!(u32_from_u8(&[]), Some(0));
+    assert_eq!(u32_from_u8(&[0xcc]), Some(0xcc));
+    assert_eq!(u32_from_u8(&[0xcc, 0x55]), Some(0xcc55));
+    assert_eq!(u32_from_u8(&[0xcc, 0x55, 0x88]), Some(0xcc5588));
+    assert_eq!(u32_from_u8(&[0xcc, 0x55, 0x88, 0xf3]), Some(0xcc5588f3));
+
+    // leading zeros are stripped
+    assert_eq!(u32_from_u8(&[0x00]), Some(0));
+    assert_eq!(u32_from_u8(&[0x00, 0x00]), Some(0));
+    assert_eq!(u32_from_u8(&[0x00, 0xcc, 0x55, 0x88]), Some(0xcc5588));
+    assert_eq!(u32_from_u8(&[0x00, 0x00, 0xcc, 0x55, 0x88]), Some(0xcc5588));
+    assert_eq!(
+        u32_from_u8(&[0x00, 0xcc, 0x55, 0x88, 0xf3]),
+        Some(0xcc5588f3)
+    );
+
+    // overflow
+    assert_eq!(u32_from_u8(&[0x01, 0xcc, 0x55, 0x88, 0xf3]), None);
+    assert_eq!(u32_from_u8(&[0x01, 0x00, 0x00, 0x00, 0x00]), None);
+    assert_eq!(u32_from_u8(&[0x7d, 0xcc, 0x55, 0x88, 0xf3]), None);
+}
+
+fn operator_lenient_mode(
+    allocator: &IntAllocator,
+    op: &[u8],
+    args: &u32,
+) -> Result<Reduction<u32>, EvalErr<u32>> {
+    if op.len() == 1 {
+        if let Some(f) = F_TABLE[op[0] as usize] {
+            let node_t: Node<IntAllocator> = Node::new(allocator, *args);
+            return f(&node_t);
+        }
+    }
+
+    // unknown opcode in lenient mode
```

Comment:
What do you think about refactoring the stuff below here out into a separate function? We could consider putting it into `more_ops` (so that we don't have to make those costs `pub`) or even into its own file.

Reply:
Also, I'm wondering about maybe just using the python version (which would require me to get to glue working), so we don't have to worry about discrepancies in implementation. Not convinced that there's much benefit to "speeding things up" in this case. Maybe this is a dumb idea.

Reply:
yeah, moving this code into more_ops makes sense to not have to touch the cost constants. Although, there may be value in consolidating those constants too, at some point. But I'll punt on that.